### PR TITLE
Unused params. 

### LINKED ISSUE #239: Pylint: W0613
```bash
mmpose/datasets/datasets/bottom_up/bottom_up_coco.py:204:0: W0613: Unused argument 'kwargs' (unused-argument)
mmpose/datasets/datasets/mesh/mesh_h36m_dataset.py:30:66: W0613: Unused argument 'logger' (unused-argument)
mmpose/utils/hooks.py:16:21: W0613: Unused argument 'model' (unused-argument)
mmpose/utils/hooks.py:16:28: W0613: Unused argument 'input' (unused-argument)
mmpose/models/detectors/bottom_up.py:266:20: W0613: Unused argument 'thickness' (unused-argument)
mmpose/models/detectors/bottom_up.py:267:20: W0613: Unused argument 'font_scale' (unused-argument)
mmpose/models/detectors/base.py:90:37: W0613: Unused argument 'optimizer' (unused-argument)
mmpose/models/detectors/base.py:90:0: W0613: Unused argument 'kwargs' (unused-argument)
mmpose/models/detectors/base.py:127:35: W0613: Unused argument 'optimizer' (unused-argument)
mmpose/models/detectors/base.py:127:0: W0613: Unused argument 'kwargs' (unused-argument)
mmpose/models/detectors/top_down.py:263:20: W0613: Unused argument 'text_color' (unused-argument)
mmpose/models/detectors/top_down.py:265:20: W0613: Unused argument 'font_scale' (unused-argument)
```

### PR DESCRIPTION
**Pull request by yaochaorui**
Fix #239

### CODE REVIEW

File: mmpose/models/detectors/base.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def _parse_losses(losses):
 
         return loss, log_vars
 
-    def train_step(self, data_batch, optimizer, **kwargs):
+    def train_step(self, data_batch):
```
Comment (dreamerlin) [inline mmpose/models/detectors/base.py:90]: The `optimizer` seems necessary for runner. https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/epoch_based_runner.py#L28-L32

Comment (innerlee) [reply mmpose/models/detectors/base.py:90]: The docstring says

>  This argument is unused and reserved.

Then let's keep it

Comment (yaochaorui) [reply mmpose/models/detectors/base.py:90]: Thanks for checking.

Comment (yaochaorui) [reply mmpose/models/detectors/base.py:90]: Updated.

**Comment by innerlee** [comment]
Since the `evaluate` methods will be refactored recently, let's close this for now

**Pull request closed without merging**